### PR TITLE
Support Scattering in Varscan, too

### LINKED ISSUE #921: scatter varscan calling in somatic WGS workflow
Right now, VarScan is a rate-limiting step, and most of that comes from the mpileup portion.  As mpileup doesn't have multi-threading, passing it a split interval bed file and scattering it (as is done with Mutect) should substantially improve performance.

### PR DESCRIPTION
**Pull request by tmooney**
* Upgrades Picard Merge VCFs to the version in GATK 4.1.8.1.
* Exposes a generic scatter parameter for all variant detectors.
* Scatters Varscan by region like other detectors.

Closes #921.

### CODE REVIEW

File: definitions/subworkflows/varscan_pre_and_post_processing.cwl [hunk 1]
```diff
@@ -49,13 +54,21 @@ outputs:
         outputSource: filter/filtered_vcf
         secondaryFiles: [.tbi]
 steps:
+    split_interval_list:
+        run: ../tools/split_interval_list.cwl
+        in:
+            interval_list: interval_list
+            scatter_count: scatter_count
+        out: [split_interval_lists]
```
[the remark this answers is not in the document]
Comment (tmooney) [reply definitions/subworkflows/varscan_pre_and_post_processing.cwl:62]: Yeah, I thought about moving it up... I kind of like having each variant detection workflow be self-contained (so you can run them standalone), and so, as you noted, we could change the scatter counts in the future if need be.  If the call caching works well it should only need to run once... but I'm not sure Cromwell can figure that out in this case.

If this step turns out to be a time/space bottleneck then I'd absolutely lean towards pulling it up.  But I figured it was relatively short and quick.  (If it's really time/space consuming we'd want to use pre-split lists as workflow inputs instead, IMO.)

All that said, I'd be happy to pull it up a level if the consensus is to prefer that!

[the remark this answers is not in the document]
Comment (chrisamiller) [reply definitions/subworkflows/varscan_pre_and_post_processing.cwl:62]: the point about being self-contained makes a lot of sense - seems fine!

**Review by chrisamiller — APPROVED** [review]
+1 looks good

File: definitions/pipelines/immuno.cwl [hunk 2]
```diff
@@ -183,7 +183,7 @@ inputs:
     strelka_cpu_reserved:
         type: int?
         default: 8
-    mutect_scatter_count:
+    scatter_count:
```
[the remark this answers is not in the document]
Comment (tmooney) [reply definitions/pipelines/immuno.cwl:186]: Excellent point!  Added doc lines... I went with `"scatters each supported variant detector (varscan, pindel, mutect) into this many parallel jobs"` though I guess that might be confusing if some workflows don't include all three of those?  I was already worried about that list drifting out of synch if a new detector is added in the future.

**Review by apaul7 — APPROVED** [review]
+1

**Pull request merged**